plugins/module_utils/structure_merging.py

```diff
@@ -0,0 +1,250 @@
+          `['A', 'B', {'C': '1', 'D': '2'}, {'E': '3'}]`
+        * and this value as `changes` :
+          `['Z', 'B', {'C': '1'}, {'E': '3'}]`
+        * if using `present=True`, get :
+          `['A', 'B', {'C': '1', 'D': '2'}, {'E': '3'}, 'Z', {'C': '1'}]`
+        * or if using `present=False`, get :
+          `['A', {'C': '1', 'D': '2'}]`
+
+        Examples when `merge_seq_by_index=True` :
+        * with this value as `base` :
+          `['A', 'B', {'C': '1', 'D': '2'}, {'E': '3'}]`
+        * and `this value as `changes` :
+          `['Z', 'B', {'C': '9', 'D': '2'}, {'E': '3'}]`
+        * if using`present=True`, get :
+          `['Z', 'B', {'C': '9', 'D': '2'}, {'E': '3'}]`
+        * or if using `present=False`, get :
+          `['A', {'C': '1'}]`
+
+        `keep_empty` parameter defines if emptied `Sequence` or `Mapping` items
+        in the structure be kept or not. This has no effect when `present`
+        parameter is `True`.
+        - When `False`, emptied `Sequence` or `Mapping are kept.
+        - When `True`, emptied `Sequence` or `Mapping` are removed.
+
+        Examples :
+        * with this value as `base` :
+          `{'A': {'AA': '1'}, 'B': ['2', '3'], 'C': '4'}`
+        * and this value as `changes` :
+          `{'A': {'AA': '1'}, 'B': ['2', '3']}`
+        * if using `keep_empty=False` with `present=False`, get :
+          `{'C': '4'}`
+        * or if using `keep_empty=True` with `present=False`, get :
+          `{'A': {}, 'B': [], 'C': '4'}`
+
+        `keep_null` parameter defines how to treat `None` value in `changes`.
+        - When `False`, `None` value are simply ignored. This can be useful
+          when is needed to only update a specific item in the `Sequence` by
+          ignoring other that be before.
+        - When `True`, all keys with in `changes` that have a `None` value are
+          removed not taking care about their value in `base`. It is useful
+          when is needed to ensure that a key will be removed.
+
+        Example that ignoring items at a specific position with
+        `remove_none=False` and `merge_seq_by_index=True` :
+        * with this value as `base` :
+          `['A', 'B', 'C', {'DA': '1', 'DB': '2'}, 'E']`
+        * and this value as `changes` :
+          `[None, None, 'Z', {'DA': '1', 'DB': None}, 'E']`
+        * if using `present=True', get :
+          `['A', 'B', 'Z', {'DA': '1', 'DB': '2'}, 'E']`
+        * or if using `present=False', get :
+          `['A', 'B', 'C', {'DB': '2'}]`
+
+        Example that remove keys/items with `remove_none=True` and `present=False` :
+        * with this value as `base` :
+         `{'A': '1', 'B': '2', 'C': {'CA': '3', 'CB': '4'}, 'D': ['DA', 'DB']}`
+        * and this value as `changes` :
+          `{'B': None, 'C': {'CB': None}, 'D': None}`
+        * get :
+          `{'A': '1', 'C': {'CA': '3'}}`
+        """
+
+        if self._both_are_assimilable_to_dict(base, changes):
```

Comment:
If I may suggest, instead of repeating this if block multiple times in the code, make a small static method that performs that logic and returns the right method to you, then here that method would be assigned to `_get_method` (or a better name as suggested before), then lines 181-187 would actually run whatever comes out, same for 194-204, 210-215, and 221-226.

For the sake of argument, will call that function `_choose_merge_method(base, change, present)`.